crates/provider/src/provider/trait.rs

```diff
@@ -537,9 +538,22 @@ pub trait Provider<T: Transport + Clone = BoxTransport, N: Network = Ethereum>:
         self.client().request("web3_clientVersion", ()).await
     }
 
-    /// Gets the chain ID.
-    fn get_chain_id(&self) -> RpcCall<T, (), U64, u64> {
-        self.client().request("eth_chainId", ()).map_resp(crate::utils::convert_u64)
+    /// Gets the chain ID. Sets the client's poll interval (if default)
+    /// to match the average block time for this chain.
+    async fn get_chain_id(&self) -> TransportResult<u64> {
+        self.client().request("eth_chainId", ()).map_resp(crate::utils::convert_u64).await.and_then(
+            |chain_id| {
+                let client = self.client();
+                if !client.is_custom_poll_interval() && !client.is_local() {
+                    let poll_interval = match Chain::from_id(chain_id).average_blocktime_hint() {
```

Comment:
this should probably do something like / 2 or `* 0.4 or 0.6` because we should use something smaller than the actual block time

Reply:
should mul by 0.6 now